analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -221,13 +215,29 @@ class GoMod(
         return graph.breakCycles()
     }
 
-    private fun ModuleInfo.toId(): Identifier =
-        Identifier(
-            type = managerName.takeIf { main } ?: "Go",
-            namespace = "",
-            name = path,
-            version = normalizeModuleVersion(version)
-        )
+    private fun ModuleInfo.toId(): Identifier {
+        val moduleInfo = replace ?: this
+
+        return if (moduleInfo.version.isBlank()) {
+            require(moduleInfo.main) {
+                "Found a local module dependency which is not supported."
+            }
+
+            Identifier(
+                type = managerName,
+                namespace = "",
+                name = moduleInfo.path,
+                version = processProjectVcs(File(moduleInfo.dir!!)).revision
+            )
+        } else {
+            Identifier(
+                type = "Go",
```

Comment:
Why is this not `"GoMod"` / `managerName`? If it's intended, please add a code comment to explain.

Reply:
And if this should be `managerName`, the code is very similar to the above clause, so it might make sense to only set a `version` conditionally, and construct the `Identifier` once in the end.